pkg/odo/cli/service/backends.go

```diff
@@ -0,0 +1,422 @@
+
+		var svcPlan scv1beta1.ClusterServicePlan
+		// if there is only one available plan, we select it
+		if len(plans) == 1 {
+			for k, v := range plans {
+				o.Plan = k
+				svcPlan = v
+			}
+			klog.V(4).Infof("Plan %s was automatically selected since it's the only one available for service %s", o.Plan, o.ServiceType)
+		} else {
+			// otherwise select the plan interactively
+			o.Plan = ui.SelectPlanNameInteractively(plans, "Which service plan should we use ")
+			svcPlan = plans[o.Plan]
+		}
+
+		o.ParametersMap = ui.EnterServicePropertiesInteractively(svcPlan)
+		o.ServiceName = ui.EnterServiceNameInteractively(o.ServiceType, "How should we name your service ", o.validateServiceName)
+		o.outputCLI = commonui.Proceed("Output the non-interactive version of the selected options")
+		o.wait = commonui.Proceed("Wait for the service to be ready")
+	} else {
+
+		o.ServiceType = args[0]
+
+		// if two args are given, first is service type and second one is service name
+		if len(args) == 2 {
+			o.ServiceName = args[1]
+		} else {
+			o.ServiceName = o.ServiceType
+		}
+
+		// we convert the param list provided in the format of key=value list
+		// to a map
+		o.ParametersMap = make(map[string]string)
+		for _, kv := range o.parameters {
+			kvSlice := strings.Split(kv, "=")
+			// key value not provided in format of key=value
+			if len(kvSlice) != 2 {
+				return errors.New("parameters not provided in key=value format")
+			}
+			o.ParametersMap[kvSlice[0]] = kvSlice[1]
+		}
+	}
+	return nil
+}
+
+func (b *ServiceCatalogBackend) ValidateServiceCreate(o *CreateOptions) (err error) {
+	// make sure the service type exists
+	classPtr, err := o.Client.GetKubeClient().GetClusterServiceClass(o.ServiceType)
+	if err != nil {
+		return fmt.Errorf("unable to create service because Service Catalog is not enabled in your cluster")
+	}
+	if classPtr == nil {
+		return fmt.Errorf("service %v doesn't exist\nRun 'odo catalog list services' to see a list of supported services.\n", o.ServiceType)
+	}
+
+	// check plan
+	plans, err := o.Client.GetKubeClient().ListMatchingPlans(*classPtr)
+	if err != nil {
+		return err
+	}
+	if len(o.Plan) == 0 {
+		// when the plan has not been supplied, if there is only one available plan, we select it
+		if len(plans) == 1 {
+			for k := range plans {
+				o.Plan = k
+			}
+			klog.V(4).Infof("Plan %s was automatically selected since it's the only one available for service %s", o.Plan, o.ServiceType)
+		} else {
+			return fmt.Errorf("no plan was supplied for service %v.\nPlease select one of: %v\n", o.ServiceType, strings.Join(ui.GetServicePlanNames(plans), ","))
+		}
+	} else {
+		// when the plan has been supplied, we need to make sure it exists
+		if _, ok := plans[o.Plan]; !ok {
+			return fmt.Errorf("plan %s is invalid for service %v.\nPlease select one of: %v\n", o.Plan, o.ServiceType, strings.Join(ui.GetServicePlanNames(plans), ","))
+		}
+	}
+	//validate service name
+	return o.validateServiceName(o.ServiceName)
```

Comment:
This functions is specific to service catalog and can be moved out of the create.go

Reply:
Moved.